clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -5102,15 +4951,13 @@ bool SYCLIntegrationFooter::emit(raw_ostream &OS) {
 // -----------------------------------------------------------------------------
 // Utility class methods
 // -----------------------------------------------------------------------------
-
-bool Util::isSyclAccessorType(QualType Ty) {
-  return isSyclType(Ty, "accessor", true /*Tmpl*/);
+bool Util::isSyclSpecialType(const QualType Ty) {
+  const CXXRecordDecl *RecTy = Ty->getAsCXXRecordDecl();
+  if (!RecTy)
+    return false;
+  return getMethodByName(RecTy, "__init");
```

Comment:
I think SYCL "special" types are identified by the attribute.
I suggest renaming this function to something like `isCXXRecordWithInitMember`.